File: python/newYearHurry.py
n, k = map(int, input().strip().split())
output = False
time = [(5 * i) for i in range(1, n+1)]
# print(time)
timeLeft = 240 - k
if timeLeft == 0:
	print(0)
else:
# print("Time Left :", timeLeft)
	for i in range(n-1, -1, -1):
		timeTaken = sum(time[0: i + 1])
		# print(f"Time taken : {timeTaken}")
		if timeTaken <= timeLeft:
			output = True
			print(i + 1)
			break
	if not output:
		print(0)

File: c++/newYearHurry.cpp
#include <iostream>
#include <numeric>
using namespace std;

int main(){
	int n, k, timeTaken = 0;
	cin >> n >> k;
	int timeLeft = 240 - k;
	for (int i = 0; i < n; ++i){
		timeTaken += 5 * (i+1);
		if (timeTaken >= timeLeft){
			if (timeTaken > timeLeft)
				cout << i;
			else if (timeTaken == timeLeft)
				cout << i + 1;
			break;
		}
	}
	if (timeTaken < timeLeft)
		cout << n;
	cout << endl;
	return 0;
}

/*
1) timeleft = 50
	timeTaken = 5 (0)
	timeTaken = 15 (1)
	timeTaken = 30 (2)
	timeTaken = 50 (3)
*/